### ISSUE TITLE
Binary classification prediction probabilities are very close to 0.5

### ISSUE THREAD

**Issue by hangyao** [Microsoft/LightGBM#925]
Hi Team,

## Environment info
Operating System: Windows 7 64-bit
CPU: Intel i7-2600 3.40GHz
C++/Python/R version: Python 3.6

## Error Message:

I run a binary classification Grid Search CV with sklearn wrapper, and the prediction probabilities of the best estimator are really close to 0.5 instead of at two ends of 0 and 1. I used the following parameters:

```
gbm = lgb.LGBMClassifier(objective='binary', 
                          n_jobs=-1, 
                          is_unbalance=True, 
                          num_threads=8, 
                          two_round=True,
                          bagging_fraction=0.9,
                          bagging_freq=1,
                          boosting_type='gbdt',
                          feature_fraction=0.9,
                          learning_rate=0.01,
                          min_child_samples=10,
                          min_child_weight=5,
                          min_data_in_leaf=20,
                          min_split_gain=0.0,
                          n_estimators=10,
                          num_leaves=21,
                          reg_alpha=0.0,
                          reg_lambda=0.0,
                          subsample=1.0)
```  

Here is the histogram and output of ```plt.hist(gbm.predict_proba(X_test)[:, 1])```:

```
(array([  3.70700000e+03,   6.79000000e+02,   1.30000000e+01,
          0.00000000e+00,   2.00000000e+00,   1.00000000e+00,
          0.00000000e+00,   0.00000000e+00,   4.96000000e+02,
          1.45600000e+03]),
 array([ 0.45398341,  0.46328985,  0.47259629,  0.48190273,  0.49120918,
         0.50051562,  0.50982206,  0.5191285 ,  0.52843494,  0.53774138,
         0.54704782]),
 <a list of 10 Patch objects>)
```

![download](https://user-images.githubusercontent.com/19369631/30656391-9ddbf63c-9df9-11e7-923e-2973553be279.png)

I believe it was caused by some hyperparameters, because using arbitrary estimator doesn't show the similar results. e.g. using ```gbm2 = lgb.LGBMClassifier(boosting_type='gbdt', num_leaves=31, n_estimators=20, objective='binary', n_jobs=-1, is_unbalance=True)``` gives the histogram of prediction probability like this, which is what I expected:

![download 1](https://user-images.githubusercontent.com/19369631/30656557-2705b13c-9dfa-11e7-9d8c-7f5f52d6d897.png)

Could you please advice if it is an error, and how to deal with that. I planned to use the prediction probability as prediction intervals or target rates for my results. If they are really close to 0.5, then it doesn't make any sense to use these values. Also it makes me wonder how LightGBM calculates prediction probability, and if using prediction probability as prediction intervals or target rates makes any sense at all even if it goes towards two ends of the histogram. 

Thank you!

**Comment by Laurae2** [comment]
You are using a learning rate of 0.01 and only 10 boosting iterations, this is an expected behavior.

**Comment by hangyao** [comment]
Thank you @Laurae2 for your prompt response! I thought the default value of num_iterations is 100? I will change the learning rate and give it a shot.

**Comment by guolinke** [comment]
@hangyao 
you set ```n_estimators=10```, so the number of iterations is 10.

**Comment by hangyao** [comment]
@guolinke Thank you! I got it. Yes, it was my fault of setting up improper parameters.
